test/package/test_directory_reader.py

```diff
@@ -0,0 +1,179 @@
+from tempfile import TemporaryDirectory
+from unittest import skipIf
+
+import torch
+from torch.package import PackageExporter, PackageImporter
+from torch.testing._internal.common_utils import (
+    run_tests,
+    IS_FBCODE,
+    IS_SANDCASTLE,
+    IS_WINDOWS,
+)
+
+try:
+    from torchvision.models import resnet18
+
+    HAS_TORCHVISION = True
+except ImportError:
+    HAS_TORCHVISION = False
+skipIfNoTorchVision = skipIf(not HAS_TORCHVISION, "no torchvision")
+
+
+try:
+    from .common import PackageTestCase
+except ImportError:
+    # Support the case where we run this file directly.
+    from common import PackageTestCase
+
+from pathlib import Path
+
+packaging_directory = Path(__file__).parent
+
+
+@skipIf(
+    IS_FBCODE or IS_SANDCASTLE or IS_WINDOWS,
+    "Tests that use temporary files are disabled in fbcode",
+)
+class DirectoryReaderTest(PackageTestCase):
+    """Tests use of DirectoryReader as accessor for opened packages."""
```

Comment:
I created a main file to consolidate the tests for this class. I can instead put the relevant tests back into their normal sections. I wasn't sure which organization system was better